apps/emqx_management/src/emqx_mgmt.erl

```diff
@@ -302,14 +302,28 @@ lookup_client({username, Username}, FormatFun) ->
     lists:append([lookup_client(Node, {username, Username}, FormatFun) || Node <- ekka_mnesia:running_nodes()]).
 
 lookup_client(Node, {clientid, ClientId}, {M,F}) when Node =:= node() ->
-    M:F(ets:lookup(emqx_channel, ClientId));
+    case ets:lookup(emqx_channel, ClientId) of
+        [] ->
+            [];
+        Ls ->
+            lists:concat(lists:map(
+              fun(Key) ->
+                case ets:lookup(emqx_channel_info, Key) of
+                    [] -> [];
+                    Ls1 -> lists:map(fun M:F/1, Ls1)
+                end
+              end, Ls))
```

Comment:
`Ls` can be `[]` for `lists:map`, so line 307 does not have to be a separate clause.